src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/TemplateRenderer.cs

```diff
@@ -109,9 +109,14 @@ public IHtmlContent Render()
 
             foreach (string viewName in GetViewNames())
             {
-                var fullViewName = modeViewPath + "/" + viewName;
+                var viewEngineResult = _viewEngine.GetView(_viewContext.ExecutingFilePath, viewName, isMainPage: false);
+                if (!viewEngineResult.Success)
+                {
+                    // Success here is more common than with GetView() but GetView() is less expensive.
```

Comment:
The pattern is flexible and we could call `FindView()` first. However 98.7% of the time, the only view engine is the `RazorViewEngine` and the difference between `GetView()` and `FindView()` is rather larger. But really the point is that the `Display(... templateName: "relative/or/absolute/path.cshtml")` cases are unusual but not enough to reverse the pattern here.

Think I may just delete the comment.